### PR TITLE
DP-3636 Add example for GeneratePostgresqlDdl

### PR DESCRIPTION
**Pull request by akmalxpert**

### CODE REVIEW

File: src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.northconcepts.datapipeline.examples.cookbook;
+
+import com.northconcepts.datapipeline.jdbc.sql.SqlPart;
+import com.northconcepts.datapipeline.sql.postgresql.*;
+
+public class GeneratePostgreSqlDdlProgrammatically {
+    public static void main(String[] args) {
+        SqlPart createUsersTable = new CreateTable("users")
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java:8]: Let's name tables using the singular -- `user`.  Please update the relevant names that reference the plural as well.

File: src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                .addColumn(new CreateTableColumn("id", TableColumnType.SERIAL).setPrimaryKey(true))
+                .addColumn(new CreateTableColumn("name", TableColumnType.VARCHAR).setNullable(false))
+                .addColumn(new CreateTableColumn("email", TableColumnType.VARCHAR).setUnique(true))
+                .addColumn(new CreateTableColumn("addressId", TableColumnType.INT))
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java]: Let's use snake case for database identifiers (and keep them lower case -- even if a particular makes them upper case behinds the scene) -- as is the norm for PostgreSQL and MySQL.

File: src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.northconcepts.datapipeline.jdbc.sql.SqlPart;
+import com.northconcepts.datapipeline.sql.postgresql.*;
+
+public class GeneratePostgreSqlDdlProgrammatically {
+    public static void main(String[] args) {
+        SqlPart createUsersTable = new CreateTable("users")
+                .addColumn(new CreateTableColumn("id", TableColumnType.SERIAL).setPrimaryKey(true))
+                .addColumn(new CreateTableColumn("name", TableColumnType.VARCHAR).setNullable(false))
+                .addColumn(new CreateTableColumn("email", TableColumnType.VARCHAR).setUnique(true))
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java]: Make sure the set a length for all `varchars`.

**Review by DeleTaylor — CHANGES_REQUESTED** [review]

File: src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java (hunk 1 above)
Comment (akmalxpert) [reply src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java:8]: it is not possible to create `user` table in PostgreSQL. that's why only this table was named in plural form.

File: src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java (hunk 2 above)
Comment (akmalxpert) [reply src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java]: done

File: src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java (hunk 3 above)
Comment (akmalxpert) [reply src/main/java/com/northconcepts/datapipeline/examples/cookbook/GeneratePostgreSqlDdlProgrammatically.java]: done

**Review by DeleTaylor — APPROVED** [review]

**Pull request merged**